packages/cli/src/credentials/credentials.service.ts

```diff
@@ -155,10 +185,25 @@ export class CredentialsService {
 			credentials = credentials.map((c) => this.ownershipService.addOwnedByAndSharedWith(c));
 		}
 
-		if (options.includeScopes) {
+		if (includeScopes) {
 			credentials = credentials.map((c) => this.roleService.addScopes(c, user, projectRelations!));
 		}
 
+		if (includeData) {
+			credentials = credentials.map((c: CredentialsEntity & ScopesField) => {
+				if (c.scopes.includes('credential:update')) {
+					return {
+						...c,
+						data: this.decrypt(c),
+					} as unknown as CredentialsEntity;
+				}
+				return {
+					...c,
+					data: undefined,
+				} as unknown as CredentialsEntity;
```

Comment:
Same here, to simplify:

```suggestion
					return {
						...c,
						data: c.scopes.includes('credential:update') ? this.decrypt(c) : undefined,
					} as unknown as CredentialsEntity;
```